Sources/Fuzzilli/Core/ProgramBuilder.swift

```diff
@@ -298,42 +306,186 @@ public class ProgramBuilder {
+        }
+
+        if let group = type.group {
+            // We check this during Environment initialization, but let's keep this just in case.
+            assert(self.fuzzer.environment.type(ofBuiltin: group) != .unknown, "We don't know how to construct \(group)")
+            let constructionSignature = self.fuzzer.environment.type(ofBuiltin: group).constructorSignature!
+            let arguments = self.generateCallArguments(for: constructionSignature)
+            let constructor = self.loadBuiltin(group)
+            obj  = self.construct(constructor, withArgs: arguments)
+        } else {
+            // Either generate a literal or use the store property stuff.
+            if probability(0.8) { // Do the literal
+                var initialProperties: [String: Variable] = [:]
+                // gather properties of the correct types
+                for prop in type.properties {
+                    let type = interpreter?.type(ofProperty: prop)
+                    var value: Variable?
+                    if type != nil && type != .unknown {
+                        value = self.randVar(ofConservativeType: type!)
+                        if value == nil {
+                            value = generateVariable(ofType: type!)
+                        }
+                    } else {
+                        value = self.randVar()
+                    }
+                    initialProperties[prop] = value
+                }
+                // TODO: This should take the method type/signature into account!
+                _ = type.methods.map { initialProperties[$0] = self.randVar(ofType: .function())! }
+                obj = self.createObject(with: initialProperties)
+            } else { // Do it with storeProperty
+                obj = self.createObject(with: [:])
+                for method in type.methods {
+                    // TODO: This should take the method type/signature into account!
+                    let methodVar = self.randVar(ofType: .function())
+                    self.storeProperty(methodVar!, as: method, on: obj)
+                }
+                // These types might have been defined in the interpreter
+                for prop in type.properties {
+                    let type = interpreter?.type(ofProperty: prop)
+                    var value: Variable?
+                    if type != nil && type != .unknown {
+                        value = self.randVar(ofConservativeType: type!)
+                        if value == nil {
+                            value = generateVariable(ofType: type!)
+                        }
+                    } else {
+                        value = self.randVar()
+                    }
+                    self.storeProperty(value!, as: prop, on: obj)
+                }
+            }
+
+            // This is needed such that generateCallArguments for later
+            // on finds this variable as the type.
+            self.setType(ofVariable: obj, to: type)
```

Comment:
Is this really needed? Shouldn't the interpreter be able to compute that type?

Reply:
Well, what's actually interesting is that I hit the assertion [here](https://github.com/carl-smith/fuzzilli-1/blob/HybridEngine/Sources/Fuzzilli/Core/ProgramBuilder.swift#L336) even after explicitly setting the type here. Not sure why that is? I think this now happens after switching from `>=` to `>` [here](https://github.com/carl-smith/fuzzilli-1/blob/HybridEngine/Sources/Fuzzilli/Core/CodeTemplate.swift#L79) but I am not entirely sure.

Reply:
Hmm, maybe add some logging to print the type that should've been generated

Reply:
Really not sure what the issue here is, I will push my recent changes and then look at this.

Reply:
ah I actually have not changed from `>=` to `>` yet.

Reply:
Maybe objects with a `group` have custom properties and if `generateVariable` does not set those as well, `randVar` fails to find them?

Reply:
Maybe, I guess you'll see that if you print the type in the assert? :)